app/src/components/guestbook/GuestBookMessageEdit.tsx

```diff
@@ -34,45 +41,93 @@ export default function GuestBookMessageEdit({
         []
     );
 
-    const updateMessage = useCallback(
-        (newValues: Partial<IGuestBookMessage>) => {
-            setMessage({
-                ...(message as IGuestBookMessage),
-                ...newValues,
-            });
-        },
-        [message, setMessage]
-    );
+
+    const initializeValues = (): NewGuestBookMessage | IGuestBookMessage => {
+        if (isUndefined(message)){
+            return getInitialMessage()
+        }
+        return message!
+    }
+
+    const validationSchema = Yup.object({
```

Comment:
This is a really nice validation schema! since it's a constant, and not a variable, can you name it like a constant (`VALIDATION_SCHEMA`) and move it outside of the component (near `PLACE_HOLDER_NAMES`)? This is more conventional for constants :)